### PR TITLE
add additional logic to check ICU version.

### LINKED ISSUE #33798: Price on product view page becomes 0,00 after page has finished loading
### Preconditions (*)

1. Magento 2.4.3
2. PHP 7.4
3. Country Options is set to US
4. Locale Options (Local) is set to Germany
5. Base Currency: Euro
6. Default Display Currency: Euro
7. Allowed Currencies: Euro
8. 3 different StoreViews are set up

### Steps to reproduce (*)

1. Create new product (price doesn't matter)
2. Go to product list page (categorie view page)
3. Go to product details page (product view page)

### Expected result (*)

1. Actual price on product details page should be shown

### Actual result (*)

1. Price on product list page is shown correctly
2. But on product details page, the page loads, correct price is shown for a millisecond but when the page has finished loading, price is shown as "0,00 €"

It seems like there is some JS or anything like that going on, when the page has finished loading, the price is set "0,00 €"
**Important: This bug is only present in 2.4.3. I downgraded to 2.4.2 and there, everything works fine.**

I found an old, _some kind of related_ issue which was fixed: https://github.com/magento/magento2/issues/4077
My store is set to **Locale Options (Local) is set to Germany**. _But I did not test if this is actually causing the issue. I will test as soon as possible._

### LINKED ISSUE #33856: Currency Symbol Disappear on Product View Page (After Fully Loaded)
### Preconditions (*)

1. Magento 2.4.3

### Steps to reproduce (*)

1. Visit any product page
2. Currency symbol appears eg - $1.00
3. Wait untill page fully load.
4. Currency symbol disappear eg - 1.00

### Expected result (*)

1. Currency Symbol to remain displayed.

### Actual result (*)

1. Currency symbol disappear

Tried with a freshly installed Magento 2.4.3. I can confirm this is a bug.
---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._

### LINKED ISSUE #33910: [Issue] Product Price Showing ZAR 0.00 when South African Currency Used
### Preconditions (*)
    Magento 2.4.3
    PHP Version 7.4.22
    Elasticsearch 7.9
    Vanilla, clean installation with sample-data

### Steps to reproduce (*)
   
1. Go to Admin -> Store -> Configuration -> Currency Setup -> Set Base Currency to South African Rand
2. Go to Admin -> Store -> Configuration -> Currency Setup -> Set Default Currency to South African Rand
3. Reindex and Flush Cache

### Expected result (*)

1. Price of the product should show as ZAR xxx (where xxx is the price).

### Actual result (*)
1. Price of the product shows as ZAR 0.00. An inspection of the value shows that ZAR 0.00 but data-price-amount is correct.
ZAR

P.S Changing back to USD shows the product price correctly.

### LINKED ISSUE #34139: Frontend Prices is 0 After upgrading from 2.4.2 to 2.4.3
### Preconditions (*)

1. Magento Version 2.4.3
2. Upgrade My Application from Version 2.4.2 to 2.4.3 Using Composer

### Steps to reproduce (*)

1. Upgrade Magento Version from 2.4.2 to 2.4.3
2. When open and product page the price appears normal then when the page completely loaded the price transferred to 0
3. When open checkout all shipping prices and product prices appears 0

Note : I install a clean Magento Version then install my themes and modules with an empty database and create some demo products till here everything is working fine but when I change the database to the main one and run the commands to upgrade it then the error mentioned in point 2 and 3 occured.

### Expected result (*)

1. Magento with Version 2.4.2 [https://prnt.sc/1t7h967](https://prnt.sc/1t7h967)

### Actual result (*)

1. Magento with Version 2.4.3 [https://prnt.sc/1t7hfzc](https://prnt.sc/1t7hfzc)

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._

### LINKED ISSUE #34229: Configurable products has zero price (in product view) after Locale is changed to Greek.
### Preconditions (*)

1. Plesk Obsidian 18.0.38
2. Magento Open Source 2.4.3
3. Php 7.4.23
4. Select English (United State) Locale from Stores --> Configuration --> General --> General (http://prntscr.com/1un7u6e)

### Steps to reproduce (*)

1. Change Catalog Input Type for Store Owner to Visual Swatch, for product attribute color (http://prntscr.com/1un5ram)
2. Create some options (http://prntscr.com/1un5v29) 
3. Add new configurable product from admin area (http://prntscr.com/1un63k5)
4. Add sample data to parent product(http://prntscr.com/1un6cec)
5. Create configurations

> - Select attribute color and click Next
> - Select all attribute values and click next
> - Skip image uploading for all SKUs
> - Apply single price to all SKUs (e.g: 50)
> - Apply single quantity to each SKUs (e.g: 120) and click Next
> - Click Generate Products

6. Save product (http://prntscr.com/1un7092)	
7. As a result the price of each product seems to be normal (http://prntscr.com/1un8fr9 , http://prntscr.com/1un8i11)
8. Change store locale from English (United State) to Greek (Greece) : http://prntscr.com/1un8o7r
9. Flush Magento Cache

### Expected result (*)

The product prices should be the same with the result of English Locale (http://prntscr.com/1un8fr9 , http://prntscr.com/1un8i11)

### Actual result (*)

All prices are zero (http://prntscr.com/1un9h1e , http://prntscr.com/1un9ii1)

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by Bashev**
### Description (*)
With this pull request add additional check for minimum version of INTL_ICU to verify compatibility with all currencies. 

### Fixed Issues (if relevant)

1. Fixes #33798 
2. Fixes #33856 
3. Fixes #33910 
4. Fixes #34139 
5. Fixes #34229

### Manual testing scenarios (*)

1. Go to random product page. 
2. See prices. 
3. If prices is greater than 0.00 and have currency, test passed.

### Questions or comments

To verify is this issue exists on your environment, check version of your ICU. If version is below 60. you may see zero prices on catalog and product pages. Issue exists on the environment. (@ihor-sviziev describes issue and steps to reproduce and verify it in #33966 )

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@Bashev would love to check where the issue reproducing, I was able to test the only on 50.2 and 65.1.
That was my guess, that issue not reproducing after version 60. Would love to see some confirmation + test results on different ICU lib versions

**Comment by Bashev** [comment]
@magento run all tests

**Comment by Bashev** [comment]
@magento run all tests

**Comment by Bashev** [comment]
@magento run Unit Tests

**Comment by Bashev** [comment]
@magento run Unit Tests

**Comment by Bashev** [comment]
@magento run Adobe CLA Signed? WebAPI Tests Static Tests

**Comment by Bashev** [comment]
@magento run Static Tests

**Comment by Bashev** [comment]
@magento run all tests

**Comment by Bashev** [comment]
@magento run Static Tests

**Comment by Bashev** [comment]
@magento run Unit Tests, Integration Tests

**Comment by Bashev** [comment]
@magento run Sample Data Tests EE, Sample Data Tests CE, Sample Data Tests B2B, Magento Health Index, Database Compare

**Comment by Bashev** [comment]
@magento run Functional Tests EE, Functional Tests CE, Functional Tests B2B, WebAPI Tests

**Comment by Bashev** [comment]
@magento run Functional Tests EE, Semantic Version Checker

File: app/code/Magento/Directory/Model/Currency.php [hunk 1]
```diff
@@ -24,6 +26,11 @@
  */
 class Currency extends \Magento\Framework\Model\AbstractModel
 {
+    /**
+     * Minimum version of ICU required.
+     */
+    const MIN_INTL_ICU_VERSION = '60.0';
```
Comment (ihor-sviziev) [inline app/code/Magento/Directory/Model/Currency.php]:
```suggestion
    private const MIN_INTL_ICU_VERSION = '60.0';
```

File: app/code/Magento/Directory/Model/Currency.php [hunk 2]
```diff
@@ -164,7 +174,7 @@ protected function _construct()
      *
      * @return string
      */
-    public function getCode()
+    public function getCode(): ?string
```
Comment (ihor-sviziev) [inline app/code/Magento/Directory/Model/Currency.php]: Please revert
```suggestion
    public function getCode()
```

File: app/code/Magento/Directory/Model/Currency.php [hunk 3]
```diff
@@ -174,7 +184,7 @@ public function getCode()
      *
      * @return string
      */
-    public function getCurrencyCode()
+    public function getCurrencyCode(): ?string
```
Comment (ihor-sviziev) [inline app/code/Magento/Directory/Model/Currency.php]:
```suggestion
    public function getCurrencyCode()
```

File: app/code/Magento/Directory/Model/Currency.php [hunk 4]
```diff
@@ -403,11 +427,11 @@ private function canUseNumberFormatter(array $options): bool
     /**
      * Format currency.
      *
-     * @param string $price
+     * @param float $price
      * @param array $options
      * @return string
      */
-    private function formatCurrency(string $price, array $options): string
+    private function formatCurrency(float $price, array $options): string
```
Comment (ihor-sviziev) [inline app/code/Magento/Directory/Model/Currency.php]: Please revert
```suggestion
    private function formatCurrency(string $price, array $options): string
```

File: app/code/Magento/Directory/Model/Currency.php [hunk 5]
```diff
@@ -403,11 +427,11 @@ private function canUseNumberFormatter(array $options): bool
     /**
      * Format currency.
      *
-     * @param string $price
+     * @param float $price
```
Comment (ihor-sviziev) [inline app/code/Magento/Directory/Model/Currency.php]:
```suggestion
     * @param string $price
```

File: app/code/Magento/Directory/Model/Currency.php [hunk 6]
```diff
@@ -560,8 +593,9 @@ public function getCurrencyRates($currency, $toCurrencies = null)
      *
      * @param array $rates
      * @return $this
+     * @throws \Magento\Framework\Exception\LocalizedException
      */
-    public function saveRates($rates)
+    public function saveRates(array $rates): Currency
```
Comment (ihor-sviziev) [inline app/code/Magento/Directory/Model/Currency.php]:
```suggestion
    public function saveRates($rates)
```

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by Bashev** [comment]
@magento run all tests

File: app/code/Magento/Directory/Model/Currency.php [hunk 7]
```diff
@@ -484,15 +511,18 @@ private function setOptions(array $options): void
...
     }
 
     /**
+     * Return Formatted Price.
+     *
      * @return string
+     * @throws \Zend_Currency_Exception
      */
-    public function getOutputFormat()
+    public function getOutputFormat(): string
```
Comment (ihor-sviziev) [inline app/code/Magento/Directory/Model/Currency.php]:
```suggestion
    public function getOutputFormat()
```

Comment (ihor-sviziev) [inline app/code/Magento/Directory/Model/Currency.php]:
```suggestion
    public function getOutputFormat()
```

File: app/code/Magento/Directory/Model/Currency.php [hunk 8]
```diff
@@ -484,15 +511,18 @@ private function setOptions(array $options): void
      *
      * @return string
      */
-    public function getCurrencySymbol()
+    public function getCurrencySymbol(): string
```
Comment (ihor-sviziev) [inline app/code/Magento/Directory/Model/Currency.php]:
```suggestion
    public function getCurrencySymbol()
```

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by Bashev** [comment]
@magento run all tests

**Comment by Bashev** [comment]
@magento run Functional Tests CE, Functional Tests B2B

**Comment by Bashev** [comment]
@magento run Functional Tests CE, Functional Tests EE

**Comment by ihor-sviziev** [comment]
@Bashev, your changes look good, but still, did you do some tests on the different ICU lib versions? Could you share some results?

**Comment by Bashev** [comment]
My test script running from magento home dir.

```
cat - <<'EOF' | php
<?php
require __DIR__ . '/app/bootstrap.php';

class TestMyClass extends \Magento\Framework\App\Http implements \Magento\Framework\AppInterface
{
    public function launch()
    {
        echo 'ICU Version: ' . INTL_ICU_VERSION . PHP_EOL;
        $formatter = new \NumberFormatter('bg_BG', \NumberFormatter::CURRENCY);
        $formatter->setSymbol(\NumberFormatter::CURRENCY_SYMBOL, '');

        /** @var \Magento\Directory\Model\Currency $currency */
        $currency = $this->_objectManager->create(\Magento\Directory\Model\Currency::class);
        echo 'Result with NumberFormatter (expected without currency symbol): ' . $formatter->formatCurrency(15.70, 'BGN') . PHP_EOL;
        echo 'Result with this PR: ' . $currency->formatTxt(15.70) . PHP_EOL;
        return $this->_response;
    }
}

$bootstrap = \Magento\Framework\App\Bootstrap::create(BP, $_SERVER);
/** @var \Magento\Framework\App\Http $app */
$app = $bootstrap->createApplication('TestMyClass');
$bootstrap->run($app);
?>
EOF
```
1. ICU Version: 65.1
✔ Result with NumberFormatter (expected without currency symbol): 15,70 
✔ Result with this PR: 15,70 лв.

2. ICU Version: 50.2
❌ Result with NumberFormatter (expected without currency symbol): 15,70 лв.
✔ Result with this PR: 15,70 лв.

3. ICU Version: 60.3
✔ Result with NumberFormatter (expected without currency symbol): 15,70 
✔ Result with this PR: 15,70 лв.

**Comment by Bashev** [comment]
Same with `fa_IR`.

```
cat - <<'EOF' | php
<?php
require __DIR__ . '/app/bootstrap.php';

class TestMyClass extends \Magento\Framework\App\Http implements \Magento\Framework\AppInterface
{
    public function launch()
    {
        echo 'ICU Version: ' . INTL_ICU_VERSION . PHP_EOL;
        $formatter = new \NumberFormatter('fa_IR', \NumberFormatter::CURRENCY);
        $formatter->setSymbol(\NumberFormatter::CURRENCY_SYMBOL, '');

        /** @var \Magento\Directory\Model\Currency $currency */
        $currency = $this->_objectManager->create(\Magento\Directory\Model\Currency::class);
        echo 'Result without PR: ' . $formatter->formatCurrency(15.70, 'BGN') . PHP_EOL;
        echo 'Result with this PR: ' . $currency->formatTxt(15.70) . PHP_EOL;
        return $this->_response;
    }
}

$bootstrap = \Magento\Framework\App\Bootstrap::create(BP, $_SERVER);
/** @var \Magento\Framework\App\Http $app */
$app = $bootstrap->createApplication('TestMyClass');
$bootstrap->run($app);
?>
EOF
```

1. ICU Version: 50.2
❌ Result with NumberFormatter (expected without currency symbol): ‎BGN۱۵٫۷۰
✔ Result with this PR: 15,70 лв.

2. ICU Version: 60.3
✔ Result with NumberFormatter (expected without currency symbol): ‎۱۶
✔ Result with this PR: 15,70 лв.

3. ICU Version: 65.1
✔ Result with NumberFormatter (expected without currency symbol): ‎۱۵٫۷۰
✔ Result with this PR: 15,70 лв.

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-9235](https://jira.corp.magento.com/browse/ENGCOM-9235) has been created to process this Pull Request

**Comment by dhorytskyi** [comment]
@Bashev Unfortunately, the issue is not fully fixed in all ICU 60+. I checked it with different ICU versions using this script:
```
foreach (['en_US', 'de_DE', 'ar_AE'] as $locale) {
    $formatter = new \NumberFormatter($locale, NumberFormatter::CURRENCY);
    $formatter->setSymbol(\NumberFormatter::CURRENCY_SYMBOL, '');
    foreach (['USD', 'EUR', 'AED'] as $currency) {
        echo 'Locale: ' . $locale . ' Currency: ' . $currency . ' ' . $formatter->formatCurrency(9.9, $currency) . PHP_EOL;
    }
}
```

ICU Version <=55
```
Locale: en_US Currency: USD $9.90
Locale: en_US Currency: EUR €9.90
Locale: en_US Currency: AED AED9.90
Locale: de_DE Currency: USD 9,90 $
Locale: de_DE Currency: EUR 9,90 €
Locale: de_DE Currency: AED 9,90 AED
Locale: ar_AE Currency: USD US$ ٩٫٩٠
Locale: ar_AE Currency: EUR € ٩٫٩٠
Locale: ar_AE Currency: AED د.إ.‏ ٩٫٩٠
```

ICU Version >=57 <=66
```
Locale: en_US Currency: USD 9.90
Locale: en_US Currency: EUR 9.90
Locale: en_US Currency: AED 9.90
Locale: de_DE Currency: USD 9,90 
Locale: de_DE Currency: EUR 9,90 
Locale: de_DE Currency: AED 9,90 
Locale: ar_AE Currency: USD ٩٫٩٠ 
Locale: ar_AE Currency: EUR ٩٫٩٠ 
Locale: ar_AE Currency: AED ٩٫٩٠ 
```

ICU Version >=67 <=69 (the latest)
```
Locale: en_US Currency: USD 9.90
Locale: en_US Currency: EUR €9.90
Locale: en_US Currency: AED AED 9.90
Locale: de_DE Currency: USD 9,90 $
Locale: de_DE Currency: EUR 9,90 
Locale: de_DE Currency: AED 9,90 AED
Locale: ar_AE Currency: USD ٩٫٩٠ US$
Locale: ar_AE Currency: EUR ٩٫٩٠ €
Locale: ar_AE Currency: AED ٩٫٩٠ 
```

Starting from ICU version 67 the issue is present again when currency code is used different from the default for the specified locale.

**Comment by dhorytskyi** [comment]
Due to the instability of the ICU library when the currency symbol is unset, I think it's better to apply approach from the previous PR, but put it in formatTxt function and use it when formatTxt is called with display = \Magento\Framework\Currency::NO_SYMBOL.

https://github.com/magento/magento2/pull/33966/files#diff-09a839078aa6453eb8fb437430724140207e2aa7214c3bb98bc7ced29fe39927R499-R501

**Comment by Bashev** [comment]
I'll try to add logic in canUseNumberFofmatter method based on output of function instead of version check.

I presume if result have no currency symbol in the result, this is enough to know, we can use NumberFormatter

**Comment by ihor-sviziev** [comment]
@dhorytskyi @Bashev ,
The issue with that solution that replace doesn't work fine in the case of fa_IR locale and USD currency:
```
cat - <<'EOF' | php
<?php

try {
    $reflector = new \ReflectionExtension('intl');
    ob_start();
    $reflector->info();
    $output = strip_tags(ob_get_clean());
    preg_match('/^ICU version (?:=>)?(.*)$/m', $output, $matches);
    $icuVersion = trim($matches[1]);
} catch (\ReflectionException $e) {
    $icuVersion = null;
}
echo 'ICU Version: ' . $icuVersion . PHP_EOL;


$formatter = new NumberFormatter('fa_IR', NumberFormatter::CURRENCY);
echo 'Original pattern: ' . $formatter->getPattern() . PHP_EOL;
echo 'Original result: ' . $formatter->formatCurrency(0, "USD") . PHP_EOL;

$formatter->setSymbol(\NumberFormatter::CURRENCY_SYMBOL, '');
echo 'Pattern with replaced currency symbol: ' . $formatter->getPattern() . PHP_EOL;
echo 'Result with changed currency symbol: ' . $formatter->formatCurrency(0, "USD") . PHP_EOL;

$formatter->setPattern( str_replace('¤','', $formatter->getPattern() ) );
echo 'Pattern with pattern: ' . $formatter->getPattern() . PHP_EOL;
echo 'Result with the changed pattern: ' . $formatter->formatCurrency(0, "USD") . PHP_EOL;

EOF
```

image

I thought about getting the "Zero" symbol and try to replace it with the `%s` as well but looks like we don't have access to the number formatter object in that context.
Any thoughts?

**Comment by Bashev** [comment]
Tested with ICU
- 50.2
- 65.1
- 69.1

btw, 2.4.3 with 69.1 works properly without fix. I presume this is because used currency is default as @dhorytskyi explain above.

**Comment by Bashev** [comment]
@magento run all tests

**Comment by Bashev** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Bashev** [comment]
@magento run all tests

**Comment by Bashev** [comment]
@magento run all tests

**Comment by Bashev** [comment]
@magento run all tests

**Comment by Bashev** [comment]
@magento run Integration Tests

**Comment by Bashev** [comment]
@magento run all tests

**Comment by engcom-Alfa** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Alfa** [comment]
Hi @Bashev and @ihor-sviziev 
Thanks for your contribution and collaboration!

I have gone through the PR and all relevant issues linked to it, tried to reproduce functionally in multiple ways and it is not getting reproduced in Magento 2.4-develop.

If we go through the comments of this [Issue](https://github.com/magento/magento2/issues/33910), it clears that the problem behind the issue is because of ICU version that is with <60 version. So the issue reproducible in AWS EC2 instance only.
Hence I will not be able to reproduce in my system.
Kindly let me know if there is any way to do as such!

cc: @sdzhepa

**Comment by engcom-Alfa** [comment]
> @engcom-Alfa same behavior is available on versions > 66.1 described in this comment by @dhorytskyi [#34156 (comment)](https://github.com/magento/magento2/pull/34156#issuecomment-930620874)

Yes Correct! But here the _php 7.4.24_ (the latest version of php) also using ICU version 66 only!
_Php 8_ is using _ICU 68_ version and that is not supported by _Magento2_! According to me the configurations will become incompatible.
Well, still trying an alternative to reproduce the issue!

**Comment by ihor-sviziev** [comment]
Hi @adifucan,
We, together with @Bashev, are working on the fix for the changes you were committing as part of the [MCP-288](https://github.com/magento/magento2/commit/c640062d97c47966e0b027f42a86774d62c4a42c).
We tried to contact you on Slack but didn't get any response from you.

More details:
https://github.com/magento/magento2/pull/33966
https://github.com/magento/magento2/pull/34156

Now we think that 2nd option is better, but I'm not sure about the performance of this change

Could you review it and give your feedback? Thank you!

**Comment by ihor-sviziev** [comment]
@engcom-Alfa,
as we said before, it's not related to the PHP version, but to the ICU library. It might be PHP 7.4 with ICU library 50.2 or 68.0. That depends on the OS requirements and the way how you're installing PHP (probably if you're installing from source - you'll  get the latest available version)

**Comment by adifucan** [comment]
Hi @ihor-sviziev @Bashev! The current approach in this PR where **Intl::NumberFormatter** is NOT used if there is a currency symbol or currency code leads to performance degradation because **Zend_Currency** is used instead. 
I believe it fixes the issue but it brings back degradation because in ~90% of cases **Zend_Currency** will be used because in most cases we do have currency symbol or code. I'll get back with numbers of performance measurements later but I see that Performance-Acceptance-Tests build is already red.

@dhorytskyi Am I correct that from ICU v.57 till ICU v.66 the issue can be fixed as the behaviour of the library is more predictable? If so, why can't we check for ICU version in **\Magento\Directory\Model\Currency::canUseNumberFormatter** like @Bashev proposed initially?
```
if (INTL_ICU_VERSION < 57 || INTL_ICU_VERSION > 66) {
    return false;
}
```

**Comment by Bashev** [comment]
closed fixed in https://github.com/magento/magento2/commit/ae304d43cd36322b1ff8145d70c9b2e9aeb0cde
@ihor-sviziev thx for support

**Pull request closed without merging**